.github/workflows/cwag-workflow.yml

```diff
@@ -58,6 +58,10 @@ jobs:
             make -C ${MAGMA_ROOT}/cwf/gateway precommit
             cd ${MAGMA_ROOT}/cwf/gateway
             make -C ${MAGMA_ROOT}/cwf/gateway/integ_tests precommit
+      - name: Check precommit has not generated formatting changes
+        run: |
+            git status
+            git diff-index --quiet HEAD
```

Comment:
Is the precommit check something that can be run locally without much effort?
* if yes, does it make sense to print the command in the failure case? this is just nit, because it can be seen in the previous step
* if no, does it make sense to show the actual diff?

Reply:
I think that's the purpose of the status command. It will show a list of changed files.

Reply:
But it is not visible, what needs to be changed. I.e., I would need to run locally
```
cd ${MAGMA_ROOT}/cwf/gateway
make -C ${MAGMA_ROOT}/cwf/gateway precommit
cd ${MAGMA_ROOT}/cwf/gateway
make -C ${MAGMA_ROOT}/cwf/gateway/integ_tests precommit
```
In order to see the changes and commit them. In github CI I see that this take ~2-3 minutes, i.e., I think it is reasonable to do this manually. But the PR owner would have to fetch these commands from the .yml file -> does it make sense to print these commands to the output in case of changes?

Reply:
The list of files is visible in the PR in case of an error (see https://github.com/magma/magma/runs/6472265618?check_suite_focus=true from the test plan).

But it might be nice to print the precommit commands that fix the issue, I agree.

Reply:
Discussed offline, @voisey it would be nice to have an additional echo for the check, something like:
echo "Checking for changes caused by make -C ${MAGMA_ROOT}/cwf/gateway precommit. Run this command locally to see respective changes."

Reply:
Nice idea, thanks! Are you happy with this commit? https://github.com/magma/magma/pull/12753/commits/5b0556e74aea9a30d4d027fa7b222f9639f3a9be. You can see how it looks in the CI here: https://github.com/magma/magma/runs/6487122263?check_suite_focus=true. If so, I can cherry pick it into this PR.

Reply:
/me is happy :) thank you!